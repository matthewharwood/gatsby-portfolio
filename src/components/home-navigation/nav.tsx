import React, { FunctionComponent } from 'react';
import { useStyletron } from 'baseui';

const Nav: FunctionComponent = ({ children }) => {
  const [css, theme] = useStyletron();
  const c = css({
    paddingTop: 0,
    paddingBottom: 0,
    paddingRight: theme.sizing.scale1400,
    paddingLeft: theme.sizing.scale1400,
    display: 'block',
    overflow: 'hidden',
  });
  return <nav className={c}>{children}</nav>;
};

export { Nav };
