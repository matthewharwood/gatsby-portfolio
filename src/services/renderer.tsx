function componentMapFactory() {
  const componentMap = {
    "postHeader": (props: any) => console.log("POSTHEADER", props),
    "summary": (props: any) => console.log("SUMMARY", props),
  }

  let defaultValueHandler = {
    get: function(target: any, name: any) {
      return target.hasOwnProperty(name)
        ? target[name]
        : (props: any) => console.warn(
          "We are missing a component type: ", props._type, "is not found"
        );
    },
  }

  return new Proxy(componentMap, defaultValueHandler)
}

export const pageRenderer = (raw) => {

  const componentMap = componentMapFactory();

  raw.map((d) => [d._type, d]).map(([key, props]) => componentMap[key](props));
  return true
}