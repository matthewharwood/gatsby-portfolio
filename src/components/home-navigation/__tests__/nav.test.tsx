import React from 'react';
import renderer from 'react-test-renderer';
import { Nav } from '../nav';

describe('Should create Nav with children', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Nav>hello</Nav>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
