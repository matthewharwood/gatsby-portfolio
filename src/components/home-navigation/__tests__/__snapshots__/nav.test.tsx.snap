// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create Nav with children renders correctly 1`] = `
<nav
  className=""
>
  hello
</nav>
`;
